var/spack/repos/builtin/packages/cryptopp/package.py

```diff
@@ -25,11 +25,29 @@ class Cryptopp(MakefilePackage):
     version('5.6.2', sha256='5cbfd2fcb4a6b3aab35902e2e0f3b59d9171fee12b3fc2b363e1801dfec53574')
     version('5.6.1', sha256='98e74d8cb17a38033354519ac8ba9c5d98a6dc00bf5d1ec3c533c2e8ec86f268')
 
+    variant('shared', default=True,  description="Build shared object versions of libraries.")
+    variant('sse',    default=True,  description="Build with instruction vectorization enabled.")
+
     depends_on('gmake', type='build')
 
     def url_for_version(self, version):
         url = '{0}/{1}{2}.zip'
         return url.format(self.homepage, self.name, version.joined)
 
+    def build(self, spec, prefix):
+        cxx_flags = []
+
+        if '+shared' in self.spec:
+            cxx_flags.append(self.compiler.cxx_pic_flag)
+        if '~sse' in self.spec:
+            cxx_flags.extend([
+                '-DCRYPTOPP_DISABLE_SSE2',
+                '-DCRYPTOPP_DISABLE_SSSE3',
+                '-DCRYPTOPP_DISABLE_SSE4',
+            ])
```

Comment:
You can do something like that (beware or typos, since I edited the suggestion directly on Github):
```suggestion
        target = self.spec.target
        if 'sse4.1' not in target:
            cxx_flag.append('-DCRYPTOPP_DISABLE_SSE4')
        if 'ssse3' not in target:
            cxx_flag.append('-DCRYPTOPP_DISABLE_SSSE3')
        if 'sse2' not in target:
            cxx_flag.append('-DCRYPTOPP_DISABLE_SSE2')
```

Reply:
This is great; thanks @alalazo! I'm certainly in agreement that fewer variants is better, and using Spack to detect when these flags should be applied is a huge boon (`cryptopp` tries on its own, but fails for at least one machine I've used). I'll add this change and run another local test presently.